Assets/LeapMotionModules/UI/Interaction/ActivityManager.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using Leap.Unity.Query;
+using UnityEngine;
+
+namespace Leap.Unity.UI.Interaction {
+
+  /// <summary> The Activity Manager encapsulates the logic that determines which
+  /// InteractionBehaviours are within an interactible distance and which interactions
+  /// they should be considered for (hovering/contact/grasping).
+  /// 
+  /// Essentially, the ActivityManager is a wrapper around PhysX sphere queries around
+  /// Hands for InteractionBehaviours.</summary>
+  public class ActivityManager {
+
+    public float activationRadius;
+    public InteractionManager manager;
+
+    private Collider[] _colliderResultsBuffer = new Collider[32];
+    private HashSet<InteractionBehaviourBase> _activeBehaviours = new HashSet<InteractionBehaviourBase>();
+
+    public HashSet<InteractionBehaviourBase> ActiveBehaviours {
+      get { return _activeBehaviours; }
+    }
+
+    public ActivityManager(InteractionManager manager, float activationRadius) {
+      this.manager = manager;
+      this.activationRadius = activationRadius;
+    }
+
+    public void FixedUpdateHand(Hand hand) {
+      int count = GetSphereColliderResults(hand, _colliderResultsBuffer, out _colliderResultsBuffer);
```

Comment:
Instead of passing in the array twice, once as an out modifier, you can just use the ref modifier for the same effect